src/plugins/intel_gna/src/ops/gna_convolution.hpp

```diff
@@ -0,0 +1,189 @@
+                   const ngraph::CoordinateDiff& pads_end,
+                   const ngraph::Strides& dilations,
+                   const ov::op::PadType& auto_pad = ov::op::PadType::EXPLICIT);
+
+    GNAConvolution(const ngraph::Output<ngraph::Node>& data_batch,
+                   const ngraph::Output<ngraph::Node>& filters,
+                   const ngraph::Strides& strides,
+                   const ngraph::CoordinateDiff& pads_begin,
+                   const ngraph::CoordinateDiff& pads_end,
+                   const ngraph::Strides& dilations,
+                   const ov::op::PadType& auto_pad = ov::op::PadType::EXPLICIT);
+
+    void validate_and_infer_types() override;
+    bool visit_attributes(ov::AttributeVisitor& visitor) override;
+
+    std::shared_ptr<ngraph::Node> clone_with_new_inputs(const ngraph::OutputVector& new_args) const override;
+
+    /// \return The strides.
+    const ngraph::Strides& get_strides() const {
+        return m_strides;
+    }
+    void set_strides(const ngraph::Strides& strides) {
+        m_strides = strides;
+    }
+    /// \return The dilations.
+    const ngraph::Strides& get_dilations() const {
+        return m_dilations;
+    }
+    void set_dilations(const ngraph::Strides& dilations) {
+        m_dilations = dilations;
+    }
+    /// \return The padding-below sizes (possibly negative).
+    const ngraph::CoordinateDiff& get_pads_begin() const {
+        return m_pads_begin;
+    }
+    void set_pads_begin(const ngraph::CoordinateDiff& pads_begin) {
+        m_pads_begin = pads_begin;
+    }
+    /// \return The padding-above sizes (possibly negative).
+    const ngraph::CoordinateDiff& get_pads_end() const {
+        return m_pads_end;
+    }
+    void set_adding_above(const ngraph::CoordinateDiff& pads_end) {
+        m_pads_end = pads_end;
+    }
+    /// \return The pad type for convolution.
+    const ov::op::PadType& get_auto_pad() const {
+        return m_auto_pad;
+    }
+    void set_auto_pad(const ov::op::PadType& auto_pad) {
+        m_auto_pad = auto_pad;
+    }
+    bool has_add_node() const {
+        return m_has_add_node;
+    }
+    ActivationType get_activation() const {
+        return m_activation_type;
+    }
+    void set_activation(ActivationType activation_type) {
+        m_activation_type = activation_type;
+    }
+
+    /*
+     * TODO: for unit tests
+    bool evaluate(ov::runtime::TensorVector& output_values,
+                  const ov::runtime::TensorVector& input_values,
+                  const ov::EvaluationContext & evaluation_context) const override;
+    bool has_evaluate() const override;
+    */
+
+protected:
+    ngraph::Strides m_strides;
+    ngraph::Strides m_dilations;
+    ngraph::CoordinateDiff m_pads_begin;
+    ngraph::CoordinateDiff m_pads_end;
+    ov::op::PadType m_auto_pad;
+    int64_t m_num_spatial = -1;
+
+private:
+    friend int64_t internal::calculate_num_spatial(const ov::intel_gna::op::GNAConvolution* op,
```

Comment:
Why do we need friend declaration here?
Just wonder if we cannot just extend interface by some accessors methods to avoid friendship declaration. F

I see that protected fields are used so probably there is plan to expose it to the subclass which suggest that inheritance will be used. Friendship is not inherited.

If we still would like to be sure that only specific method  or class have access to this method we could could use PassKey idiom as it was done in https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_gna/src/request/model_wrapper.hpp#L23